catalog/ui/catalog-ui-search/src/main/webapp/js/CQLUtils.js

```diff
@@ -98,7 +98,9 @@ define([
             var cqlPolygon = model.map(function (point) {
                 return point[0] + ' ' + point[1];
             });
-            cqlPolygon.push(cqlPolygon[0]);
+            if (cqlPolygon[0] !== cqlPolygon[cqlPolygon.length - 1]) {
+                cqlPolygon.push(cqlPolygon[0]);
```

Comment:
No, it adds a duplicate of the first point on to the end if the first point does not match the last point.  This ensures the polygon is closed.